File: adventofcode/cpp/year2019/day11/day11_test.cc
#include "adventofcode/cpp/year2019/day11/day11.h"

#include <fstream>
#include <string>

#include "gtest/gtest.h"

#include "adventofcode/cpp/adventofcode.h"

TEST(Year2019Day11, Part1Actual) {
  std::ifstream input("adventofcode/data/year2019/day11/actual.in");
  std::string output = "2093";
  adventofcode::answer_t a = day11::part1(input);
  EXPECT_EQ("", a.error);
  EXPECT_EQ(output, a.answer);
  input.close();
}

TEST(Year2019Day11, Part2Actual) {
  std::ifstream input("adventofcode/data/year2019/day11/actual.in");
  std::ifstream output_file("adventofcode/cpp/year2019/day11/testdata/out");
  std::string output(std::istreambuf_iterator<char>(output_file), {});
  adventofcode::answer_t a = day11::part2(input);
  EXPECT_EQ("", a.error);
  EXPECT_EQ(output, a.answer);
  input.close();
}

File: adventofcode/java/year2016/day11/Day11Test.java
package adventofcode.java.year2016.day11;

import java.io.FileReader;
import java.io.IOException;

import org.junit.Test;
import org.junit.Assert;

public class Day11Test {
    @Test
    public void part1Example() throws IOException {
        try (var input = new FileReader("adventofcode/java/testdata/year2016/day11/example")) {
            var output = "11";
            var result = Day11.part1(input);
            Assert.assertEquals("no error", "", result.error);
            Assert.assertEquals("correct output", output, result.answer);
        }
    }

    @Test
    public void part1Actual() throws IOException {
        try (var input = new FileReader("adventofcode/data/year2016/day11/actual.in")) {
            var output = "31";
            var result = Day11.part1(input);
            Assert.assertEquals("no error", "", result.error);
            Assert.assertEquals("correct output", output, result.answer);
        }
    }

     @Test
     public void part2Actual() throws IOException {
         try (var input = new FileReader("adventofcode/data/year2016/day11/actual.in")) {
             var output = "55";
             var result = Day11.part2(input);
             Assert.assertEquals("no error", "", result.error);
             Assert.assertEquals("correct output", output, result.answer);
         }
     }
}
